### ISSUE TITLE
Regression on generated schemas between 1.28 and 1.29

### ISSUE THREAD

**Issue by jpjpjp** [apideck-libraries/portman#630]
Hi. I think I've stumbled upon a regression in the generated schema in the contract tests. This appears to have been introduced in 1.29.

I've got an API that has its response defined as follows:
```yaml
      responses:
        "200":
          description: A list of Category Objects
          content:
            application/json:
              schema:
                type: object
                properties:
                  categories:
                    type: array
                    items:
                      oneOf:
                        - $ref: "#/components/schemas/categoryObject"
                        - $ref: "#/components/schemas/categoryGroupObject"
```

So essentially, the API returns an object with a property called `categories`, which is an array of objects which can be one of two types.

In Portman 1.28, the schema generated in the contract test for this API looks like this (somewhat abbreviated for simplicity):

```json
{
  "type": "object",
  "properties": {
    "categories": {
      "type": "array",
      "items": {
        "oneOf": [
          {
            "type": "object",
            "additionalProperties": false,
            "properties": {
              "id": {
                "type": "integer",
                "format": "int64",
                "nullable": false,
                "description": "A system defined unique identifier for the category."
              },
              "name": {
                "type": "string",
                "nullable": false,
                "description": "The name of the category.",
                "minLength": 1,
                "maxLength": 100
              },
              "is_group": {
                "type": "boolean",
                "enum": [
                  false
                ],
                "description": "For categoryObject, is_group is always false."
              }
            },
            "required": [
              "id",
              "name",
              "is_group"
            ]
          },
          {
            "type": "object",
            "properties": {
              "id": {
                "type": "integer",
                "format": "int64",
                "nullable": false,
                "description": "A system defined unique identifier for the category."
              },
              "name": {
                "type": "string",
                "nullable": false,
                "description": "The name of the category.",
                "minLength": 1,
                "maxLength": 100
              },
              "is_group": {
                "type": "boolean",
                "enum": [
                  true
                ],
                "description": "For categoryGroupObject, is_group is always true."
              },
              "group_id": {
                "type": [
                  "integer",
                  "null"
                ],
                "format": "int64",
                "description": "The ID of the category group this category belongs to or `null` if the category doesn't belong to a group, or is itself a category group."
              },
              "children": {
                "type": "array",
                "items": {
                  "type": "integer"
                },
                "description": "For category groups, this will populate with the child category IDs."
              }
            },
            "required": [
              "id",
              "name",
              "is_group"
            ]
          }
        ] 
      }
    }
  }
}
```
So, the key thing to notice about this (correct) schema is that it captures the OneOf property.

Here's the (simplified) schema generated by 1.29.0:
```json
{
  "type": "object",
  "properties": {
    "categories": {
      "type": "array",
      "items": {
        "type": "object",
        "additionalProperties": false,
        "properties": {
          "id": {
            "type": "integer",
            "format": "int64",
            "nullable": false,
            "description": "A system defined unique identifier for the category."
          },
          "name": {
            "type": "string",
            "nullable": false,
            "description": "The name of the category.",
            "minLength": 1,
            "maxLength": 100
          },
          "is_group": {
            "type": "boolean",
            "enum": [
              false
            ],
            "description": "For categoryObject, is_group is always false."
          }
        },
        "required": [
          "id",
          "name",
          "is_group"
        ]
      }
    }
  }
}
```
Notice that this generated schema does not have a `oneOf` property. It expects all objects in the returned array to match the first schema in the oneOf list.

I hope this makes sense. @thim81, I'm not ready to share the full spec publicly, but I've added you to a private repo that will allow you to grab it and reproduce it if you are able to.

**Comment by thim81** [comment]
@jpjpjp I'll look into this and see what we can find as a resolution.

**Comment by thim81** [comment]
hi @jpjpjp 

We have found the cause of the OAS object not being Immutable and have a [PR](https://github.com/apideck-libraries/portman/pull/636) ready for review.

**Comment by thim81** [comment]
hi @jpjpjp 

We just released Portman 1.29.3, which contains a fix for the JSON schema issue you faced.
Let us know if it your issue is solved, by closing the issue.

If there is another regression, please do keep reporting it.

**Comment by jpjpjp** [comment]
Happy to confirm that v1.29.3 addresses the issue I originally reported, and continues to generate the rest of my tests as expected.

Thank you!!
